### PR TITLE
Re 6866/create symlinks

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
This will use custom actions in the MSI to create symlinks in the INSTALLDIR/bin to the binaries in INSTALLDIR/puppet/bin.

NOTE:: This PR will not remove the existing shortcuts in that directory, that part will be done by: https://github.com/puppetlabs/puppet-agent/pull/607

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
This has been tested on 2012r2 x64 and it works there

File: resources/windows/wix/customactions.wxs.erb [hunk 1]
```diff
@@ -212,5 +212,67 @@ End If
...
+    <CustomAction
+    Id="PuppetShellBatLink"
+    Execute="immediate"
+    ExeCommand="&quot;[%WINDIR]\System32\cmd.exe&quot; /c mklink /H &quot;[INSTALLDIR]bin\puppet_shell.bat&quot; &quot;[INSTALLDIR]puppet\bin\puppet_shell.bat&quot;"
+    Directory="ShortCutBinDir"/>
+
+    <CustomAction
+    Id="RunPuppetInteractiveBatLink"
+    Execute="immediate"
+    ExeCommand="&quot;[%WINDIR]\System32\cmd.exe&quot; /c mklink /H &quot;[INSTALLDIR]bin\run_puppet_interactive.bat&quot; &quot;[INSTALLDIR]puppet\bin\run_puppet_interactive.bat&quot;"
```
Comment (ferventcoder) [inline resources/windows/wix/customactions.wxs.erb]: These are all hardlinks. Was this what you were looking for?

**Comment by ferventcoder** [comment]
We are going to explore not needing to do custom actions. They should be a last resort when there are no other options.

**Comment by ferventcoder** [comment]
Closing for now.

**Pull request closed without merging**